---
title: "RNAseq plab6"
author: "Kelterborn"
date: "2024-03-15"
output:
  md_document:
  variant: gfm
toc: true
always_allow_html: true
editor_options: 
  chunk_output_type: console
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=300)
```

# 0. Prepare System
BiocManager::install()

## Load System
```{r library}
library(stringr)
library(R.utils)
library(RColorBrewer)
library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)
library(tximeta)
library(tximport)
library(curl)
library(AnnotationHub)
library(DESeq2)
library(EnhancedVolcano)
library(pheatmap)
library(writexl)
library(biomaRt)
library(ape)
library(kableExtra)
library(knitr)

library(stringr)
library(R.utils)
library(RColorBrewer)

library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)
library(tximeta)
library(tximport)
library(curl)

library(SummarizedExperiment)
library(GenomicRanges)
library(ape)

library(viridis)
library(patchwork)
library("ggpubr")
library(vsn)

library(PCAtools)

# library(wget)

```
## Download data
Username: uFZjGr6t
Password: nYGTAdUw
Link https://filetransfer.mdc-berlin.de/?u=uFZjGr6t&p=nYGTAdUw

You can find detailed instruction accessing the folder in different settings here: https://t1p.de/gdmhowtocrush 

```{r folders}

# wget("https://filetransfer.mdc-berlin.de/?u=uFZjGr6t&p=nYGTAdUw")

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/P3043",sep="/")
gitdir <- paste(dir,"git_Chlamy_RNAseq_plap6",sep="/")
datadir <- paste(dir,"data",sep="/")
outdir <- gitdir
quantdir <- paste(dirname(datadir),"quants2",sep="/")

dir1 <- paste(dirname(dir),"linux-ngs/salmon/salmon_index_chlamy",sep="/")

indexDir <- file.path(dirname(dir),"linux-ngs/salmon/salmon_index_chlamy/chlamy_index_v6.1")
fastaPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.0.hardmasked.fa.gz")
# list.files(dirname(fastaPath))
# head(readLines(fastaPath,n=50))
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.repeatmasked_assembly_v6.0.gff3.gz")
# readLines(gtfPath,n=20)
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.gene_exons.gff3.gz")
# readLines(gtfPath,n=20)
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.gene.gff3.gz")
# readLines(gtfPath,n=20)

file.exists(indexDir, fastaPath, gtfPath)

```
setwd("S:/AG/AG-Scholz-NGS/Daten/Simon/P3043")

# 1. Process Data
## Sample names
```{r sample_table}

f <- list.files(path = quantdir)
files <- as.factor(str_remove(f,pattern ="_quant"))
# short form
sample <- str_remove(files,pattern ="P3043_") 
sample %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

sampleid <- {}
strain  <- {}
lane  <- {}
condition <- {}
media  <- {}
replicate <- {}
trial <- {}

for (i in list.files(path = quantdir)){
  # print(i)
  si <- substring(i,14,18)
  sampleid <-c(sampleid,str_split(i, pattern = "_", simplify = TRUE)[,5])
  condition.i <- str_split(i, pattern = "_", simplify = TRUE)[,4]
  strain <-c(strain,str_split(condition.i, pattern = "-", simplify = TRUE)[,1])
  media <-c(media,str_split(condition.i, pattern = "-", simplify = TRUE)[,2])
  replicate <-c(replicate,str_split(condition.i, pattern = "-", simplify = TRUE)[,3])
  lane <-c(lane,str_split(i, pattern = "_", simplify = TRUE)[,6])
}
strain <- str_replace_all(strain, pattern="fib",replacement="Δplap6")
condition <- paste(strain,media,sep="_") %>% factor(levels = c("WT_TAP","Δplap6_TAP","WT_HSM","Δplap6_HSM"))
strain  <- strain %>% factor() %>% relevel("WT")
lane  <- lane %>% factor()
media  <- media %>% factor()  %>% relevel("TAP")
replicate <- replicate %>% factor()

trial <- c(rep(1,6),rep(2,8),rep(1,2),rep(2,4),rep(1,8),rep(1,12),rep(2,12),rep(1,12)) %>% factor() %>% relevel("1")

# outliner: S15, S16, S17, S18, S19, S26



# Make table
sample.table <- data.frame(sample,sampleid,strain,media,condition,replicate,trial,lane)

sample.table$samplename <- paste(condition,sampleid,sep="_")
outlier <- c("S15", "S16", "S17", "S18", "S19", "S26")
sample.table$outlier <- str_detect(sample.table$sampleid, paste(outlier, collapse = "|")) %>% factor() %>% relevel("FALSE")

sample.table %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
sample.table$strain %>% levels()

```


## Mapping Rates

```{r mapping_rates}

# list.files(path = outdir)
# list.files(path = datadir)
# list.files(path = quantdir)

samplelist <- {}
mappingrates <- {}
for (i in list.files(path = quantdir)){
  # print(i)
  si <- paste("",str_sub(i,11,-7),sep = "")
  samplelist <- c(samplelist,si)
  f <- readLines(paste(quantdir,i,"logs/salmon_quant.log", sep="/"))
  line <- grep("Mapping rate = ",f,value=TRUE)
  sl <- str_length(line)
  notime <- substring(line,30,sl)
  manual <- substring(line,sl-7,sl-1)
  val <- as.numeric(str_extract(notime,"[0-9.]+"))
  valr<-round(val, digits=2)
  # print(paste("Mapping rate of ",si," is: ",valr," %"))
  mappingrates <- c(mappingrates,valr)
}

# Make table

m.table <- data.frame(sample.table,mappingrates)
m.table[1:10,c("sample","mappingrates")] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
```

### Plot mapping rates
```{r plot mappingrates, fig.height=10}

# Colours

# Plot
# plot(m.table$mappingrates)
# -> boring

# increase margin for longer names
# par(mar=c(4,15,4,4)+.1)
# barplot(height=mappingrates, names=m.table$sample, horiz=T, las=1)

par(mar=c(4,8,4,4)+.1)
xx <- barplot(height=mappingrates, names=m.table$sample, cex.names=0.5, horiz=T, las=1, xlim=c(0,100)) #col=col
text(x = mappingrates, y = xx, label = mappingrates, pos = 4, cex = 0.5, col = "red")


```

## Tximeta

```{r tximeta, eval=FALSE}

example.quant <- paste(quantdir,list.files(quantdir)[1],"quant.sf",sep="/")

list.files(quantdir)[1]
file.exists(example.quant)

# first file:
example.table <- read.table(example.quant, header=T)
head(example.table)
basename(example.quant)
dirname(example.quant)
basename(dirname(example.quant))

# generate file list & prepare variables
files={}
for (i in list.dirs(path = dirname(dirname(example.quant)), full.names = TRUE, recursive = FALSE)) {
  files <- c(files,paste(i,"/quant.sf",sep=""))
  # print(basename(i))
  # print(head(read.table(files[length(files)], header=T)))
  # print (files[length(files)])
}
# head(files)

dir <- dirname(files)
# head(dir)
# head(basename(dir))
run <- basename(dir)
dirdir <- dirname(dir)
# head(dirdir)

tximeta_files <- file.path(dirdir, run, "quant.sf") 
file.exists(files,tximeta_files) %>% summary()

# names as working sample names
# see 'm.table' at mapping rates
coldata <- data.frame(files, names = run, stringsAsFactors=FALSE)
coldata <- data.frame(coldata,m.table)
coldata %>% head(10) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
colnames(coldata)
condition %>% levels()

# load tximeta
# with linked Transcriptome
## Chlamy

file.exists(indexDir, fastaPath, gtfPath)

makeLinkedTxome(indexDir=indexDir,
                source="Phytozome",
                organism="Chlamydomonas reinhardtii",
                release="v6.1",
                genome="CreinhardtiiCC_4532_707",
                fasta=fastaPath,
                gtf=gtfPath,
                write=FALSE)

gtfdata <- readLines(gtfPath)
# head(gtfdata, n=20)


# gene_name info sind in gff3 abder nicht in se?

se <- tximeta(coldata, useHub=TRUE)
se

colData(se) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# meta infos sind da (genotype, media,...)

# rename Samples
rownames(colData(se)) <- colData(se)$sample.id

rowRanges(se)
# genome info
seqinfo(se)
# ?
head(assays(se)[["counts"]])
# counts. THE DATA

# Mapping infos:
# names(metadata(se)[["quantInfo"]])
# str(metadata(se)[["quantInfo"]]) # Infos from Salmon Mapping
metadata(se)[["quantInfo"]]$percent_mapped
metadata(se)[["quantInfo"]]$num_processed

par(mfrow=c(1,2),
    mar=c(4,4,4,4)+.1)
barplot(metadata(se)[["quantInfo"]]$percent_mapped, main="Mapping Rate")
barplot(metadata(se)[["quantInfo"]]$num_processed/1000000, main="Mio. Reads")
par(mfrow=c(1,1))
edb <- retrieveDb(se)
class(edb)
genes(edb) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
columns(edb)


se.exons <- addExons(se)
rowRanges(se.exons)[[1]] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
gse <- summarizeToGene(se)
rowRanges(gse) %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

head(assays(gse)[["counts"]])[1:5,1:5] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
head(assays(gse)[["abundance"]])[1:5,1:5] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
head(assays(gse)[["length"]])[1:5,1:5] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

#### add gene symbol
columns(retrieveDb(se))
edb
metadata(se)$txomeInfo$source

TXNAME <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "TXNAME", keytype = "GENEID", multiVals="first"))
# select()' returned 1:many mapping between keys and columns = 1 gene has many transcripts...
TXNAME.list <- (mapIds(edb,keys = mcols(gse)$gene_id, column = "TXNAME", keytype = "GENEID", multiVals="list"))
head(TXNAME.list)
# info is already there
CDSID <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "CDSID", keytype = "GENEID", multiVals="first"))
head(CDSID)
# ?
CDSNAME <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "CDSNAME", keytype = "GENEID", multiVals="first"))
head(CDSNAME)
# same as gene_id
EXONNAME <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "EXONNAME", keytype = "GENEID", multiVals="first"))
head(EXONNAME)
TXTYPE <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "TXTYPE", keytype = "GENEID", multiVals="first"))
head(TXTYPE)
# no useful info... no gene Symbol!!

# add CDSID
mcols(gse)$CDSID <- as.character(mapIds(edb,keys = mcols(gse)$gene_id, column = "CDSID", keytype = "GENEID", multiVals="first"))


colnames(mcols(gse))
head(rownames(gse))

# getwd()
# datadir
# saveRDS(gse, file = paste(datadir,"gse.RDS",sep="/"), compress = FALSE)
# gse <- {}
# gse <- readRDS(paste(datadir,"gse.RDS",sep="/"))
# gse

```

## Get Gene Names from GFF3 file
### a-I) Prepare gene.gff3 data
Use data from transcriptome used fore mapping: CreinhardtiiCC_4532_707_v6.1.gene.gff3.gz

```{r gene.gff3, eval=FALSE}
readLines(gtfPath,n=10)
gff <- read.delim(gtfPath, header=F, comment.char="#")
head(gff)

getAttributeField <- function (x, field, attrsep = ";") {
  s = strsplit(x, split = attrsep, fixed = TRUE)
  sapply(s, function(atts) {
    a = strsplit(atts, split = "=", fixed = TRUE)
    m = match(field, sapply(a, "[", 1))
    if (!is.na(m)) {
      rv = a[[m]][2]
    }
    else {
      rv = as.character(NA)
    }
    return(rv)
  })
}

gffRead <- function(gffFile, nrows = -1) {
  cat("Reading ", gffFile, ": ", sep="")
  gff = read.table(gffFile, sep="\t", as.is=TRUE, quote="",
                   header=FALSE, comment.char="#", nrows = nrows,
                   colClasses=c("character", "character", "character", "integer",
                                "integer",
                                "character", "character", "character", "character"))
  colnames(gff) = c("seqname", "source", "feature", "start", "end","score", "strand", "frame", "attributes")
  cat("found", nrow(gff), "rows with classes:",
      paste(sapply(gff, class), collapse=", "), "\n")
  stopifnot(!any(is.na(gff$start)), !any(is.na(gff$end)))
  return(gff)
}

Chlamy_gff <- gffRead(gtfPath)
Chlamy_gff
Chlamy_gff$Name <- getAttributeField(Chlamy_gff$attributes, "Name")
Chlamy_gff$ID <- getAttributeField(Chlamy_gff$attributes, "ID")
# Chlamy_gff$Parent <- getAttributeField(Chlamy_gff$attributes, "Parent")
Chlamy_gff$pacid <- getAttributeField(Chlamy_gff$attributes, "pacid")
# Chlamy_gff$tID <- getAttributeField(Chlamy_gff$attributes, "tID")
Chlamy_gff$geneName <- getAttributeField(Chlamy_gff$attributes, "geneName")
# Chlamy_gff$longest <- getAttributeField(Chlamy_gff$attributes, "longest")
dim(Chlamy_gff)

# all IDs, some Gene symbols but no description etc...

# get only genes with Gene symbol
Chlamy_genes <- subset(Chlamy_gff, !geneName =="")
Chlamy_genes$gene_id <- paste(str_split(Chlamy_genes$Name, pattern = "_", simplify = TRUE)[,1])
gene2symbol <- data.frame(gene_id=Chlamy_genes$gene_id,
                          Symbol=Chlamy_genes$geneName)
dim(gene2symbol)
gene2symbol[duplicated(gene2symbol$gene_id),]
gene2symbol <- gene2symbol[!duplicated(gene2symbol$gene_id),]
dim(gene2symbol)
summary(mcols(gse)$gene_id %in% gene2symbol$gene_id)
# all FALSE (Cre01.g050800_4532)
summary(gene2symbol$gene_id %in% mcols(gse)$gene_id)
# 1 outliner

gene2symbol[!gene2symbol$gene_id %in% mcols(gse)$gene_id,] %>% dim()
# CreCp.g802322	icreI	

```


### a-II) add gene.gff3 data into gse
```{r add_gff3, eval=FALSE}

# Remove "_4532"
mcols(gse)$gene_id2 <- mcols(gse)[,"gene_id"]
mcols(gse)$gene_id <- str_remove(mcols(gse)[,"gene_id2"], pattern = "_4532")
rownames(gse) <- mcols(gse)$gene_id

gsetable <- as.data.frame(mcols(gse)[,c("gene_id","gene_id2")])

dim(gsetable)
dim(gene2symbol)

summary(mcols(gse)$gene_id %in% gene2symbol$gene_id)
summary(gene2symbol$gene_id %in% mcols(gse)$gene_id)

new <- plyr::join(gsetable,gene2symbol)
dim(new)
head(new)
new2 <- left_join(gsetable,gene2symbol,by = "gene_id")
head(new2)

summary(new == new2)
# both methods worked!

mcols(gse)$symbol <- new$Symbol
mcols(gse)

# saveRDS(gse, file = paste(datadir,"gse.RDS",sep="/"), compress = FALSE)
# gse <- readRDS(paste(datadir,"gse.RDS",sep="/"))
# gse

```


### b-I) More gene data: master_annotation_table.tsv
Use other files from Phytozome database: CreinhardtiiCC_4532_707_v6.1.master_annotation_table.tsv

```{r master_annotation, eval=FALSE}

# more gene info from master_annotation_table
annopath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.master_annotation_table.tsv")

anno <- read.delim2(annopath, header = T, sep = "\t",)
head(anno) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# search for genes
anno[str_detect(anno[["locusName_4532"]],"Cre01.g000050"),]
anno[str_detect(anno[["geneSymbol"]],"HKR"),]
anno[str_detect(anno[["previousIdentifiers"]],"HKR"),]
anno[str_detect(anno[["geneSymbol"]],"COP"),]
anno[str_detect(anno[["previousIdentifiers"]],"COP1"),]
anno[str_detect(anno[["geneSymbol"]],"cop"),]
anno[str_detect(anno[["geneSymbol"]],"PLAP6"),]
anno[str_detect(anno[["geneSymbol"]],"COQ3"),]
anno[str_detect(anno[["geneSymbol"]],"PCRY"),]
anno[str_detect(anno[["geneSymbol"]],"CRY"),]
anno[str_detect(anno[["geneSymbol"]],"CRY-DASH"),]
anno[str_detect(anno[["previousIdentifiers"]],"CRY-DASH"),]

dim(anno)

# Remove "_4532"
anno$gene_id <- str_remove(anno$locusName_4532,pattern = "_4532")
rownames(anno) <- anno$gene_id

# convert previousIdentifiers to list
anno$previousIdentifiers_list <- strsplit(anno$previousIdentifiers, "#")
head(anno) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno$previousIdentifiers_list[1]

# polish "previousIdentifiers"
# remove Cre & g-numbers
head(anno$previousIdentifiers)
tail(anno$previousIdentifiers)
prev.symbols <-  
  str_replace_all(anno$previousIdentifiers,pattern="Cre\\d+.g\\d+.t\\d+.\\d+", replacement="?") %>% 
  str_replace_all(pattern="Cre...g\\d+", replacement="?") %>% 
  str_replace_all(pattern="g\\d+.t\\d+", replacement="?") %>% 
  str_remove_all(pattern="\\?") %>%
  str_remove_all("^#|#$| ") %>% str_remove_all("^#|#$") %>% str_remove_all("^#|#$|^0") %>% 
  str_replace_all(pattern="##","#") %>% str_replace_all(pattern="##","#")
prev.symbols[str_detect(prev.symbols,"#")] %>% head(n=20)
prev.symbols[str_detect(prev.symbols,"0")] %>% head(n=20)
prev.symbols[str_detect(prev.symbols,"Cre")] %>% head(n=20)
tail(anno$previousIdentifiers[str_detect(anno$previousIdentifiers,"Cre")])


anno$prev.symbols <- prev.symbols
head(anno, n=10) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# Make colum with gene symbol or prev.symbol or Gene id
anno$combname <- anno$geneSymbol

summary(anno$combname)
summary(anno$combname=="")
summary(anno$combname==" ")
summary(is.na(anno$combname))
summary(anno$combname=="0")

# replace "" with prev.symbol
summary(anno$combname=="")
length(anno$combname[anno$combname==""])
length(str_split(
  anno$prev.symbols[anno$combname==""],pattern ="#", simplify = T)[,1])
anno$combname[anno$combname==""] <- str_split(
  anno$prev.symbols[anno$combname==""],pattern ="#", simplify = T)[,1]
summary(anno$combname=="")

anno$combname[anno$combname==""] <- anno$gene_id[anno$combname==""]
summary(anno$combname=="")
subset(anno,anno$combname=="")

summary(duplicated(anno$combname))
# some duplicates

colnames(anno)[which(names(anno) == "combname")] <- "id.symbol"


summary(anno$geneSymbol!="")
# 4408 gene Symbols
summary(anno$previousIdentifiers!="")
# all (!) 17712 genes have "previousIdentifiers"
summary(anno$prev.symbols!="")
# 7236 have alternative/old symbols
summary(anno$id.symbol)

dim(anno)

saveRDS(anno, file = paste(datadir,"anno.RDS",sep="/"), compress = FALSE)
anno <- readRDS(paste(datadir,"anno.RDS",sep="/"))

```


### b-II) add anno data into gse ####
```{r add_anno, eval=FALSE}
anno <- readRDS(paste(datadir,"anno.RDS",sep="/"))

summary(anno$gene_id %in% mcols(gse)$gene_id)
summary(mcols(gse)$gene_id %in% anno$gene_id)

# choose info
anno_join <- anno[,c("gene_id", "geneSymbol", "id.symbol","prev.symbols", "previousIdentifiers", "previousIdentifiers_list", "Description", "Comments", "TargetP", "Predalgo")]

class(gsetable)
class(anno_join)

dim(gsetable)
dim(anno_join)

# any duplicated:
summary(duplicated(anno$gene_id))

new <- plyr::join(gsetable,anno_join, by = "gene_id", type = "left")
dim(new)
head(new)
new2 <- left_join(gsetable,anno_join,by = "gene_id")
dim(new2)
head(new2)

mcols(gse) <- left_join(as.data.frame(mcols(gse)),anno_join,by = "gene_id")
mcols(gse)

getwd()

saveRDS(gse, file = paste(datadir,"gse.RDS",sep="/"), compress = FALSE)
gse <- readRDS(paste(datadir,"gse.RDS",sep="/"))
gse

```


## DESeq2 Analysis
```{r desq2, fig.show="hold", out.width="50%"}

gse <- readRDS(paste(datadir,"gse.RDS",sep="/"))

colData(gse) %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
head(mcols(gse)) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
gse$condition %>% levels()

# dds <- DESeqDataSet(gse, ~media+strain+media:genotype)
dds <- DESeqDataSet(gse, ~condition)
colData(dds)$names %>% head()

##########################################\n
## filter all rows with rowsum = 0 ####

hist(log(counts(dds)), breaks=100, ylim = c(0,100000))
hist(counts(dds), breaks=1000000, ylim = c(0,100000), xlim = c(0,100))
# -> dip at log(counts(dds)=2)

# remove 0 counts
remove0 <- rowSums(counts(dds) == 0) == length(colData(dds)[, 1])
summary(remove0)
dds <- dds[!remove0,]

# min 16 counts in 16 samples
sample.number <- nrow(colData(dds)) / colData(gse)$condition %>% levels() %>% length()
keep.sn <- rowSums(counts(dds) >= sample.number) >= sample.number
summary(keep.sn)
dds <- dds[keep.sn,]

dds
head(assays(dds)[["counts"]])[1:5,1:5] %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

length(rownames(counts(dds)))

## combine technical replicates #####
dds <- collapseReplicates(dds, dds$sampleid,dds$lane)
colData(dds) %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
dds$runsCollapsed
colnames(dds)
matchFirstLevel <- dds$sampleid == levels(factor(dds$sampleid))[1]
all(rowSums(counts(dds[,matchFirstLevel])) == counts(dds[,1]))

# dds <- estimateSizeFactors(dds)
colData(dds)$genotype <- colData(dds)$strain


###########
# run DESeq
dds <- DESeq(dds)
plotMA(dds)

plotCounts(dds, gene = "Cre10.g423750", intgroup = "condition", col=colData(dds)$outlier, main = "Cre10.g423750 (COQ3)")

###############
# save dds #####

# saveRDS(dds, file = paste(datadir,"dds.RDS",sep="/"), compress = FALSE)
# dds <- readRDS(paste(datadir,"dds.RDS",sep="/"))

```


## -Quality I
### - Data transformations
```{r pre_trans}
# dds <- readRDS(paste(datadir,"dds.RDS",sep="/"))
vsd <- vst(dds, blind=FALSE) #Variance stabilized transformation
ntd <- normTransform(dds)
```

```{r pre_trans_fig, fig.show="hold", out.width="33%"}
meanSdPlot(assay(ntd))
meanSdPlot(assay(vsd))

```

### - Check sample distance
```{r pre_sample_dist, fig.height=9, out.width="100%"}
sampleDists <- dist(t(assay(vsd)))

sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- vsd$samplename
# colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
anno_col <- as.data.frame(colData(vsd)[,c("strain","media","trial","outlier")])
anno_colors <- list(media = c("yellow","skyblue1"),
                    strain = c("seagreen2","orange"),
                    trial = c("grey90","grey30"),
                    outlier = c("black","red"))

names(anno_colors$media) <- levels(anno_col$media)
names(anno_colors$strain) <- levels(anno_col$strain)
names(anno_colors$trial) <- levels(anno_col$trial)
names(anno_colors$outlier) <- levels(anno_col$outlier)

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         annotation_col=anno_col,
         annotation_colors = anno_colors,
         show_colnames     = FALSE,
         col=viridis(20),
         cutree_rows = 8,
         cutree_cols = 8)

```

### - Advanced PCA
```{r pca_advanced, warning = FALSE, fig.show="hold", out.width="80%", fig.width=8}
vst_dat <- assay(vst(dds))
p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.2)

bi1 <- biplot(p,x="PC2",y="PC1",
    lab = p$metadata$experiment,
    colby = 'condition',colkey = viridis(4),
    hline = 0, vline = 0,
    encircle = TRUE, encircleFill = TRUE,
    legendLabSize = 10, legendIconSize = 4.0,
    legendPosition = 'bottom')
bi2 <- biplot(p,x="PC2",y="PC1",
    lab = p$metadata$experiment,
    colby = 'outlier',colkey = c("grey30","red1"),
    hline = 0, vline = 0,
    encircle = TRUE, encircleFill = TRUE,
    legendLabSize = 10, legendIconSize = 4.0,
    legendPosition = 'bottom')
bi1+bi2

p3 <- pairsplot(p,colby = 'outlier', colkey = c("grey30","red1"),title = 'outlier',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # -> PC4 = experiment
p4 <- pairsplot(p,colby = 'condition', colkey = viridis(4),title = 'condition',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # PC1 & PC3 = condition

p3+p4

plotloadings(p,
             components = getComponents(p)[1:7],
             col = viridis(3),labSize = 3)
```




##
vst_dat <- assay(vst(dds))
p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.2)




## Remove outlier
# outliner: S15, S16, S17, S18, S19, S26

```{r remove_outlier, fig.show="hold", out.width="50%"}

outliner <- c("S15", "S16", "S17", "S18", "S19", "S26")
rmo <- colnames(dds)[str_detect(colnames(dds), paste(outliner, collapse = "|"))]
keep <- colnames(dds)[!str_detect(colnames(dds), paste(outliner, collapse = "|"))]
keep

dds[,keep]
dds <- dds[,keep] 

# run DESeq
dds <- DESeq(dds)
plotMA(dds)

plotCounts(dds, gene = "Cre10.g423750", intgroup = "condition",col=colData(dds)$trial)

# save dds #####

saveRDS(dds, file = paste(datadir,"dds.RDS",sep="/"), compress = FALSE)
dds <- readRDS(paste(datadir,"dds.RDS",sep="/"))
dds

```


## -Quality II
### - Data transformations
```{r pre_trans2}
dds <- readRDS(paste(datadir,"dds.RDS",sep="/"))
vsd <- vst(dds, blind=FALSE) #Variance stabilized transformation
ntd <- normTransform(dds)
```

```{r pre_trans_rlog2, eval=TRUE}
rld <- rlog(dds, blind=FALSE) #regularized logarithm
save(rld,file=paste(datadir,"rlog.rld", sep="/"))
rld <- 1
load(file=paste(datadir,"rlog.rld", sep="/"))
rld
```

```{r pre_trans_fig2, fig.show="hold", out.width="33%"}
load(file=paste(datadir,"rlog.rld", sep="/"))
meanSdPlot(assay(ntd))
meanSdPlot(assay(vsd))
meanSdPlot(assay(rld))

```

### - Check sample distance
```{r pre_sample_dist2, fig.width=9, out.width="100%"}
sampleDists <- dist(t(assay(vsd)))

sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- vsd$samplename
# colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
anno_col <- as.data.frame(colData(vsd)[,c("strain","media","trial","outlier")])
anno_colors <- list(media = c("yellow","skyblue1"),
                    strain = c("seagreen2","orange"),
                    trial = c("grey90","grey30"),
                    outlier = c("black","red"))

names(anno_colors$media) <- levels(anno_col$media)
names(anno_colors$strain) <- levels(anno_col$strain)
names(anno_colors$trial) <- levels(anno_col$trial)
names(anno_colors$outlier) <- levels(anno_col$outlier)

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         annotation_col=anno_col,
         annotation_colors = anno_colors,
         show_colnames     = FALSE,
         col=viridis(20),
         cutree_rows = 8,
         cutree_cols = 8)

```

### - Advanced PCA
```{r pca_advanced2, warning = FALSE, fig.show="hold", out.width="80%", fig.width=8}
vst_dat <- assay(vst(dds))
p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.9)

bi1 <- biplot(p,x="PC2",y="PC1",
    lab = p$metadata$experiment,
    colby = 'condition',colkey = viridis(4),
    hline = 0, vline = 0,
    encircle = TRUE, encircleFill = TRUE,
    legendLabSize = 10, legendIconSize = 4.0,
    legendPosition = 'bottom')
bi2 <- biplot(p,x="PC2",y="PC1",
    lab = p$metadata$experiment,
    colby = 'trial',colkey = c("grey30","red1"),
    hline = 0, vline = 0,
    encircle = TRUE, encircleFill = TRUE,
    legendLabSize = 10, legendIconSize = 4.0,
    legendPosition = 'bottom')
bi1+bi2

# check different PCAs
p1 <- pairsplot(p,colby = 'media', colkey = viridis(2),title = 'media',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # -> PC1 = treatment
p2 <- pairsplot(p,colby = 'genotype', colkey = viridis(2),title = 'genotype',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # -> PC1&3 = genotype
p3 <- pairsplot(p,colby = 'trial', colkey = c("grey30","red1"),title = 'trial',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # -> PC4 = experiment
p4 <- pairsplot(p,colby = 'condition', colkey = viridis(4),title = 'condition',titleLabSize = 15,trianglelabSize = 12,
                hline = 0, vline = 0,gridlines.major = FALSE, gridlines.minor = FALSE,
                plotaxes = FALSE,margingaps = unit(c(-0.01, -0.01, -0.01, -0.01), 'cm')) # PC1 & PC3 = condition

p1+p2+p3+p4

plotloadings(p,
             components = getComponents(p)[1:7],
             col = viridis(3),labSize = 3)
```






