---
title: "Chlamy_plap6_dive"
author: "Kelterborn"
date: "2024-03-15"
output:
  md_document:
  variant: gfm
toc: true
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
    })    
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=300)
```

# 0. Start
## Load System
```{r library}
library(stringr)
library(R.utils)
library(RColorBrewer)
library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)
library(tximeta)
library(tximport)
library(curl)
library(AnnotationHub)
library(DESeq2)
library(EnhancedVolcano)
library(pheatmap)
library(writexl)
library(biomaRt)
library(ape)
library(kableExtra)
library(knitr)

library(stringr)
library(R.utils)
library(RColorBrewer)

library(sessioninfo)
library(data.table)
library(plyr)
library(tidyverse)
library(tximeta)
library(tximport)
library(curl)

library(SummarizedExperiment)
library(GenomicRanges)
library(ape)

library(viridis)
library(patchwork)
library("ggpubr")
library(vsn)

library(PCAtools)
library(aplot)

# library(wget)

```

### R folders
```{r folders}

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/P3043",sep="/")
gitdir <- paste(dir,"git_Chlamy_RNAseq_plap6",sep="/")
datadir <- paste(dir,"data",sep="/")
outdir <- gitdir
pubdir <- paste(gitdir,"pub_figures",sep="/")
quantdir <- paste(dirname(datadir),"quants2",sep="/")

dir1 <- paste(dirname(dir),"linux-ngs/salmon/salmon_index_chlamy",sep="/")

indexDir <- file.path(dirname(dir),"linux-ngs/salmon/salmon_index_chlamy/chlamy_index_v6.1")
fastaPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.0.hardmasked.fa.gz")
# list.files(dirname(fastaPath))
# head(readLines(fastaPath,n=50))
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.repeatmasked_assembly_v6.0.gff3.gz")
# readLines(gtfPath,n=20)
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.gene_exons.gff3.gz")
# readLines(gtfPath,n=20)
gtfPath <- file.path(dir1,"Phytozone_v6.1/CreinhardtiiCC_4532_707_v6.1.gene.gff3.gz")
# readLines(gtfPath,n=20)

file.exists(indexDir, fastaPath, gtfPath)

```

## Load dds
```{r load_dds}
dds <- readRDS(paste(datadir,"dds.RDS",sep="/"))
anno <- readRDS(paste(datadir,"anno.RDS",sep="/"))

# load(paste(datadir,"dds.RDS",sep="/"))

```

## Make results 

```{r make_results, fig.show="hold", out.width="50%"}
resultsNames(dds)

# TAP = Δplap6_TAP - WT_TAP
# HSM = Δplap6_HSM - WT_HSM
# Δplap6 = Δplap6_HSM - Δplap6_TAP
# WT = WT_HSM - WT_TAP

res_Δplap6.TAP.vs.WT.TAP <-
  results(dds, contrast = c("condition","Δplap6_TAP","WT_TAP"))
res_Δplap6.TAP.vs.WT.TAP$Symbol <- mcols(dds)$id.symbol

res_Δplap6.HSM.vs.WT.HSM <-
  results(dds, contrast = c("condition","Δplap6_HSM","WT_HSM"))
res_Δplap6.HSM.vs.WT.HSM$Symbol <- mcols(dds)$id.symbol

res_Δplap6.HSM.vs.Δplap6.TAP <-
  results(dds, contrast = c("condition","Δplap6_HSM","Δplap6_TAP"))
res_Δplap6.HSM.vs.Δplap6.TAP$Symbol <- mcols(dds)$id.symbol

res_WT.HSM.vs.WT.TAP <-
  results(dds, contrast = c("condition","WT_HSM","WT_TAP"))
res_WT.HSM.vs.WT.TAP$Symbol <- mcols(dds)$id.symbol

res1 <- res_Δplap6.TAP.vs.WT.TAP
res2 <- res_Δplap6.HSM.vs.WT.HSM
res3 <- res_Δplap6.HSM.vs.Δplap6.TAP
res4 <- res_WT.HSM.vs.WT.TAP

summary(res1)
top_res1 <- subset(res1, padj < 0.01 & baseMean > 50 &
                     (log2FoldChange < -1 | log2FoldChange > 1))
top_res1 <- top_res1[order(top_res1$log2FoldChange, decreasing = T),]
dim(top_res1)

summary(res2)
top_res2 <- subset(res2, padj < 0.01 & baseMean > 50 &
                     (log2FoldChange < -1 | log2FoldChange > 1))
top_res2 <- top_res2[order(top_res2$log2FoldChange, decreasing = T),]
dim(top_res2)

summary(res3)
top_res3 <- subset(res3, padj < 0.01 & baseMean > 50 &
                     (log2FoldChange < -2 | log2FoldChange > 2))
top_res3 <- top_res3[order(top_res3$log2FoldChange, decreasing = T),]
dim(top_res3)

summary(res4)
top_res4 <- subset(res4, padj < 0.01 & baseMean > 50 &
                     (log2FoldChange < -2 | log2FoldChange > 2))
top_res4 <- top_res4[order(top_res4$log2FoldChange, decreasing = T),]
dim(top_res4)

resLFC_TAP <- lfcShrink(dds, contrast = c("condition","Δplap6_TAP","WT_TAP"), type="ashr")
resLFC_TAP$Symbol <- mcols(dds)$id.symbol
resLFC_HSM <- lfcShrink(dds, contrast = c("condition","Δplap6_HSM","WT_HSM"), type="ashr")
resLFC_HSM$Symbol <- mcols(dds)$id.symbol
resLFC_Δplap6 <- lfcShrink(dds, contrast = c("condition","Δplap6_HSM","Δplap6_TAP"), type="ashr")
resLFC_Δplap6$Symbol <- mcols(dds)$id.symbol
resLFC_WT <- lfcShrink(dds, contrast = c("condition","WT_HSM","WT_TAP"), type="ashr")
resLFC_WT$Symbol <- mcols(dds)$id.symbol

res1["Cre01.g000150",]
plotMA(res1, main = "Δplap6 vs. WT in TAP", ylim=c(-4,4))
plotMA(resLFC_TAP, main = "Δplap6 vs. WT in TAP", ylim=c(-4,4))
summary(res1)

```


# 2.) Data Dive

## Get gen names

```{r gene_names}
# Load data
COQ3 <- "Cre10.g423750"
PHO1 <- "Cre08.g359300"

# search genes
anno[str_detect(anno[["geneSymbol"]],"COQ3"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
# get only gene IDs
anno[str_detect(anno[["geneSymbol"]],"CRY"),"gene_id"] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# gene ID
# g245450
anno[str_detect(anno[["gene_id"]],"g245450"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")


# More searches:
anno[str_detect(anno[["geneSymbol"]],"ROC"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno[str_detect(anno[["prev.symbols"]],"CRY"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno[str_detect(anno[["Description"]],paste(c('Opsin', 'opsin','photoreceptor'), collapse="|")),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno[str_detect(anno[["Comments"]],paste(c('phototaxis'), collapse="|")),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# all Chloroplast
anno %>% filter_at(.vars = vars(Predalgo, TargetP),
                   .vars_predicate = any_vars(str_detect
                                              (. , paste0("^(", paste("Chloroplast", collapse = "|"), ")")))) %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
# all Flagellar
anno[str_detect(anno[["Flagellar_Proteome"]],"Total Peptides:"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# all transmembrane with 7 helices
anno[str_detect(anno[["TMHMM_transmembrane"]],"TMHMM: 7 helices"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

# fibrillin related
coqs <- anno[str_detect(anno[["geneSymbol"]],"COQ"),]
anno[str_detect(anno[["Comments"]],paste(c('Coenzyme Q'), collapse="|")),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
goi <- anno[c(PHO1,coqs[,"gene_id"]),]

```

### Genes Yousef 
```{r goi}
#
tocopherol <- c("Cre09.g414000", "Cre01.g013801", "Cre14.g624350", "Cre09.g393400")
plastoquinone <- c("Cre09.g398993", "Cre06.g283750", "Cre14.g625450", "Cre12.g503550")
phylloquinone <- c("Cre10.g455950", "Cre16.g671000", "Cre04.g219787", "Cre09.g416500")
goi_y <- c(tocopherol,plastoquinone, phylloquinone)
goi_y

```


## Plot Counts
### Examples
```{r countsexamples}
head(mcols(dds)$symbol)
anno[str_detect(anno[["geneSymbol"]],"CRY-DASH"),]
# no hit
g <- anno[str_detect(anno[["previousIdentifiers"]],"CRY-DASH1"),"gene_id"]
g
plotCounts(dds, gene = g, intgroup = "condition")
g <- anno[str_detect(anno[["geneSymbol"]],"PCRY"),"gene_id"]
plotCounts(dds, gene = g, intgroup = "condition", col=colData(dds)$genotype, main =anno[g,"geneSymbol"])
g <- anno[str_detect(anno[["geneSymbol"]],"ACRY"),"gene_id"]
plotCounts(dds, gene = g, intgroup = "condition", col=colData(dds)$genotype, main =anno[g,"geneSymbol"])
g <- anno[str_detect(anno[["geneSymbol"]],"ROC15"),"gene_id"]
plotCounts(dds, gene = g, intgroup = "condition", col=colData(dds)$genotype, main =anno[g,"geneSymbol"])
g <- anno[str_detect(anno[["geneSymbol"]],"ROC40"),"gene_id"]
plotCounts(dds, gene = g, intgroup = "condition", col=colData(dds)$genotype, main =anno[g,"geneSymbol"])


# search genes
anno[str_detect(anno[["geneSymbol"]],"CRY"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno[str_detect(anno[["geneSymbol"]],"ROC"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
anno[str_detect(anno[["previousIdentifiers"]],"CRY"),1:9] %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

gene <- COQ3
plotCounts(dds, gene = gene, intgroup = "condition", col=colData(dds)$genotype, main =anno[gene,"geneSymbol"])

# with ggplot
d <- plotCounts(dds, gene = gene, intgroup = c("condition","media","genotype"), col=colData(dds)$genotype, main =anno[gene,"geneSymbol"], returnData=TRUE)
d %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")
# extract data and save as table

g1 <- ggplot(d, aes(x = condition, y = count, color = genotype)) + 
  geom_point() +
  ggtitle(anno[gene,"geneSymbol"])
plot(g1)

# more advanced:
g1 <- ggplot(d, aes(x = media, y = count, color = genotype)) + 
  geom_boxplot(aes(group = condition, colour = genotype)) +
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  # geom_text_repel(aes(label = rownames(d))) +
  scale_y_continuous(trans = "log2") +
  theme_bw() +
  ggtitle(anno[gene,"geneSymbol"]) +
  theme(plot.title = element_text(hjust = 0.5))
plot(g1)
```

### COQs
```{r counts_coqs, fig.show="hold",out.width="33%"}

# all COQ genes
goi <- anno[c(PHO1,coqs[,"gene_id"]),]
goi

n <- {0}
for (i in goi$gene_id){
  n <- n+1
  print(n)
  print(i)
  s <- print(anno[i,"geneSymbol"])
  d <-  plotCounts(dds, gene=i, intgroup=c("condition","media","genotype"),main=s,returnData=TRUE)
  g <- ggplot(d, aes(x = media, y = count, color = genotype)) + 
    geom_boxplot(aes(fill=genotype), alpha=0.5) +
    geom_point(position=position_dodge(width=0.75)) +
    scale_y_continuous(trans = "log2",limits = c(250,2100)) +
    # coord_cartesian(ylim = c(0,2500)) +
    scale_color_manual(values=c("grey30","orchid1")) +
    scale_fill_manual(values=c("grey30","orchid1")) +
    theme_bw() +
    ggtitle(s) +
    theme(plot.title = element_text(hjust = 0.5)) +
    ylab("Normalized read counts [log2])")
  assign(x = paste("g",n, sep=""),g)
  plot(g)
}

length(goi)

p <- g9+g2+g11+g6+g8+g12+ plot_layout(guides = "collect", axis_titles="collect", axes = 'collect_y') & theme(legend.position = 'right')
p


# ga <- ggarrange(g1,g2,g3,g4,g5,g6,g7,g8,g9,g10,g11,g12,ncol = 3, nrow = 4)
# ga1 <- ggarrange(g2+rremove("xlab"),
#                 g8+rremove("xlab")+rremove("ylab"),
#                 g6+rremove("xlab")+rremove("ylab"),
#                 g12+rremove("xlab"),
#                 g11+rremove("xlab")+rremove("ylab"),
#                 g9+rremove("xlab")+rremove("ylab"),
#                 common.legend = TRUE, legend = "right",
#                 widths = c(1.05,1,1), heights = c(1,1),
#                 ncol = 3, nrow = 2)
# ga1
# ggexport(ga1, filename = "pub_figures/Counts_COQs.pdf",width = 8.2, height = 4.7)
# ggsave(ga1, filename = "pub_figures/Counts_COQs.png",width = 8.2, height = 4.7)

# ((g2+g8+g6)/(g12+g11+g9)) + plot_layout(guides = "collect", axes = "collect", axis_titles="collect")


```

#### -export COQs
```{r export coqs, eval=FALSE}
ggexport(p, filename = paste(pubdir,"Counts_plap6_COQs.pdf",sep="/"),width = 8.2, height = 4.7)
ggsave(p, filename = paste(pubdir,"Counts_plap6_COQs.tiff",sep="/"),width = 8.2, height = 4.7)

```




### CIA5, rbcl, LHCSR3, LHCSR1, PSBS
```{r counts_cia5_rbcl, fig.show="hold",out.width="33%"}
anno[str_detect(anno[["geneSymbol"]],"CCM1"),1:9]
anno[str_detect(anno[["previousIdentifiers"]],"CIA5"),1:9]
anno[str_detect(anno[["geneSymbol"]],"PSBS"),1:9]


PLAP6 <- "Cre03.g188700"
CIA5 <- "Cre02.g096300"
rbcl <- "CreCp.g802313"
LHCSR3 <- c("Cre08.g367400","Cre08.g367500")
LHCSR1 <- "Cre08.g365900"
PSBS <- c("Cre01.g016600","Cre01.g016750")

## multiple genes
# all COQ genes
goi <- anno[c(CIA5,LHCSR3,LHCSR1,PSBS),]  # ,rbcl
goi[,c("geneSymbol","Description")]
goi$geneSymbol[1] <- "CIA5"



n <- {0}
g_list <- list()
for (i in goi$gene_id){
  n <- n+1
  print(n)
  print(i)
  s <- print(goi[i,"geneSymbol"])
  d <-  plotCounts(dds, gene=i, intgroup=c("condition","media","genotype"),main=s,returnData=TRUE)
  g <- ggplot(d, aes(x = media, y = count, color = genotype)) + 
    geom_boxplot(aes(fill=genotype), alpha=0.5) +
    geom_point(position=position_dodge(width=0.75)) +
    scale_y_continuous(trans = "log2") +
    # coord_cartesian(ylim = c(0,2500)) +
    scale_color_manual(values=c("grey30","orchid1")) +
    scale_fill_manual(values=c("grey30","orchid1")) +
    theme_bw() +
#    ggtitle(paste0(s," (",goi[i,"gene_id"],")")) +
    ggtitle(s) +
    theme(plot.title = element_text(hjust = 0.5)) + 
    ylab("Normalized read counts [log2])")
  g_list[[s]] <- g
  assign(x = paste("g",n, sep=""),g)
  plot(g)
}
length(goi)

g_list
patchwork::wrap_plots(g_list) + plot_layout(guides = "collect", axis_titles="collect", axes='collect_y')

p <- g1 + g2 + plot_layout(guides = "collect", axis_titles="collect", axes='collect')
p


# only rbcl in TAP
i <- goi$gene_id[2]
d <-  plotCounts(dds, gene=i, intgroup=c("condition","media","genotype"),main=s,returnData=TRUE)

gt <- ggplot(d[d$media == "TAP",], aes(x = media, y = count, color = genotype)) + 
    geom_boxplot(aes(fill=genotype), alpha=0.5) +
    geom_point(position=position_dodge(width=0.75)) +
    scale_y_continuous(trans = "log2") +
    # coord_cartesian(ylim = c(0,2500)) +
    scale_color_manual(values=c("grey30","orchid1")) +
    scale_fill_manual(values=c("grey30","orchid1")) +
    theme_bw() +
    ggtitle(s) +
    theme(plot.title = element_text(hjust = 0.5))
plot(gt)


```

#### -export CIA5
```{r export cia5, eval=FALSE}
ggexport(g1, filename = paste(pubdir,"Counts_plap6_CIA5.pdf",sep="/"),width = 8.2, height = 4.7)
ggsave(g1, filename = paste(pubdir,"Counts_plap6_CIA5.tiff",sep="/"),width = 8.2, height = 4.7)

ggexport(gt, filename = paste(pubdir,"Counts_plap6_rbcl.pdf",sep="/"),width = 8.2, height = 4.7)
ggsave(gt, filename = paste(pubdir,"Counts_plap6_rbcl.tiff",sep="/"),width = 8.2, height = 4.7)

ggexport(p, filename = paste(pubdir,"Counts_plap6_CIA5+rbcl.pdf",sep="/"),width = 8.2, height = 4.7)
ggsave(p, filename = paste(pubdir,"Counts_plap6_CIA5+rbcl.tiff",sep="/"),width = 8.2, height = 4.7)

```



### PQs
```{r plotcounts2}
library("ggpubr")

## multiple genes
goi <- anno[goi_y,]
dim(goi)

n <- {0}
for (i in goi$gene_id){
  n <- n+1
  print(n)
  print(i)
  s <- print(anno[i,"id.symbol"])
  print
  d <-  plotCounts(dds, gene=i, intgroup=c("condition","media","genotype"),main=s,returnData=TRUE)
  g <- ggplot(d, aes(x = condition, y = count, color = genotype)) + 
    geom_boxplot(aes(group = condition, colour = genotype)) +
    geom_point() + #position=position_jitter(w = 0.1,h = 0)
    # geom_text_repel(aes(label = rownames(d))) + 
    theme_bw() +
    ggtitle(s) +
    theme(plot.title = element_text(hjust = 0.5))
  assign(x = paste("g",n, sep=""),g)
  plot(g)
}
length(goi)
ga2 <- cowplot::plot_grid(g1,g2,g3,g4,g5,g6,g7,g8,g9,g10,g11,g12, byrow = FALSE, nrow = 4, ncol = 3)
ga2

# ggexport(ga, filename = "graphs3/Counts_YYK.pdf",width = 12, height = 12)

```


## Volcano Plot

```{r volcano}
library(EnhancedVolcano)

plot(res1$log2FoldChange,res1$baseMean )

# colnames(mcols(dds))
# mcols(dds) <- left_join(as.data.frame(mcols(dds)),anno[,c("gene_id","prev.symbols","id.symbol")],by = "gene_id")

summary(res1)

EnhancedVolcano(res1,
                x = 'log2FoldChange',
                y = 'pvalue',
                lab = mcols(dds)$id.symbol,
                # boxedLabels = TRUE,
                labSize = 2.0,
                drawConnectors = TRUE,
                max.overlaps = 20,
                # xlim = c(-10, 10),
                # ylim = c(0, 50),
                ylab = "Padj (-log10)",
                title = 'Δplap6 vs. WT in TAP',
                subtitle = "DE genes: ?? (red)",
                # sub = "SVF",
                pCutoff = 0.01,
                FCcutoff = 2,
                pointSize = 2.0,
                legendLabels=c('Not sig.','|L2F| > 2','p-adj < 0.01',
                               'p-adj & L2F'),
                legendPosition = 'right',
)

# ggsave("graphs3/EnhancedVolcano.pdf",
#        width = 12,
#        height = 10)

# with ggplot
ggplot(as.data.frame(res1),aes(log2FoldChange,-log(padj))) +
  geom_point() +
  geom_point(data = as.data.frame(res1[rownames(top_res1),]), colour = "blue") + 
  geom_point(data = as.data.frame(res1[goi$gene_id,]), colour = "red") +
  geom_text_repel(data = as.data.frame(res1[goi$gene_id,]), aes(label=anno[goi$gene_id,"id.symbol"]),colour = "red",hjust=-0.5, vjust=-1) + xlim(-5,5)

# ggsave("graphs3/Vulcano COQs-2.pdf",
#        width = 10,
#        height = 6)

```


## Heatmap
```{r heatmap1}
library("pheatmap")
ntd <- normTransform(dds)

# select top 100 highest expressed genes
select <- order(rowMeans(counts(dds,normalized=TRUE)),
                decreasing=TRUE)[1:100]
df <- as.data.frame(colData(dds)[,c("condition","media","genotype")])
pheatmap(assay(ntd)[select,], cluster_rows=TRUE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df)

# all top genes
select <- unique(
  rownames(top_res1),
  rownames(top_res2)) %>% 
  unique(rownames(top_res3)) %>%
  unique(rownames(top_res4)) %>%
  unique(rownames(goi))

length(select)
df <- assay(ntd)[select,]
rownames(df) <- mcols(dds)[select,"id.symbol"]

anno_col <- as.data.frame(colData(dds)[,c("condition","media","genotype")])
xx <- pheatmap(df, cluster_rows=TRUE, show_rownames=TRUE,
               cluster_cols=TRUE, annotation_col=anno_col)
# ggsave("graphs3/Heatmap_top.pdf",plot=xx,
#        width = 10,
#        height = 30)

# all fib genes
select <- unique(c(
  rownames(top_res1),
  rownames(top_res2),
  rownames(goi)))

select <- unique(c(head(rownames(top_res1),n=50),
                   tail(rownames(top_res1),n=50),
                   head(rownames(top_res2),n=50),
                   tail(rownames(top_res2),n=50),
                   rownames(goi)))

length(select)

df <- assay(ntd)[select,]
rownames(df) <- mcols(dds)[select,"id.symbol"]

anno_col <- as.data.frame(colData(dds)[,c("condition","media","genotype")])
xx <- pheatmap(df, cluster_rows=TRUE, show_rownames=TRUE,
               cluster_cols=TRUE, annotation_col=anno_col)
# ggsave("graphs3/Heatmap_Δplap6.vs.WT.pdf",plot=xx,
#        width = 10,
#        height = 30)

```

### HM PQs
```{r heatmap_PQ}
library("pheatmap")
ntd <- normTransform(dds)

df <- assay(ntd)[goi$gene_id,]
rownames(df) <- mcols(dds)[goi$gene_id,"id.symbol"]

anno_col <- as.data.frame(colData(dds)[,c("condition","media","genotype")])
xx <- pheatmap(df, cluster_rows=TRUE, show_rownames=TRUE,
               cluster_cols=TRUE, annotation_col=anno_col)

# ggsave("graphs3/Heatmap_YYK.pdf",plot=xx,
#        width = 10,
#        height = 10)

```



# Export

```{r export, eval=FALSE}
library(writexl)

# combine results

colnames(anno)

res_exp <- data.frame(
  # general
  "gene_id" = rownames(dds),
  "gene_name" = anno[rownames(dds),"geneSymbol"],
  "Alias" = anno[rownames(dds),"prev.symbols"],
  "id.symbol" = anno[rownames(dds),"id.symbol"],
  "Description" = anno[rownames(dds),"Description"],
  "Comments" = anno[rownames(dds),"Comments"],
  "TargetP" = anno[rownames(dds),"TargetP"],
  "Predalgo" = anno[rownames(dds),"Predalgo"],
  "baseMean" = res1$baseMean,
  # results 1
  "L2FC.Δplap6_TAP.vs.WT_TAP" = res1$log2FoldChange,
  "pvalue.Δplap6_TAP.vs.WT_TAP" = res1$pvalue,
  "padj.Δplap6_TAP.vs.WT_TAP" = res1$padj,
  # results 2
  "L2FC.Δplap6_HSM.vs.WT_HSM" = res2$log2FoldChange,
  "pvalue.Δplap6_HSM.vs.WT_HSM" = res2$pvalue,
  "padj.Δplap6_HSM.vs.WT_HSM" = res2$padj,
  # results 3
  "L2FC.WT_HSM.vs.WT_TAP" = res3$log2FoldChange,
  "pvalue.WT_HSM.vs.WT_TAP" = res3$pvalue,
  "padj.WT_HSM.vs.WT_TAP" = res3$padj,
  # results 4
  "L2FC.Δplap6_HSM.vs.Δplap6_TAP" = res4$log2FoldChange,
  "pvalue.Δplap6_HSM.vs.Δplap6_TAP" = res4$pvalue,
  "padj.Δplap6_HSM.vs.Δplap6_TAP" = res4$padj,
  counts(dds, normalized = TRUE))
res_exp %>% head() %>% kable() %>% kable_styling("striped", full_width = T) %>% scroll_box(height = "400px")

write_xlsx(data.frame(res_exp),
           paste(outdir,"2023_08 P3043 results.xlsx",sep="/"))

write_xlsx(data.frame(colData(dds)),
           paste(outdir,"2023_08 P3043 samples.xlsx",sep="/"))


```


# End
```{r}

```


